tests/gbs/test_similarity.py

```diff
@@ -194,3 +194,46 @@ def test_sample_max_count_deterministic(self, photon_num, count, monkeypatch):
             m.setattr("numpy.random.choice", lambda x: x[0])
             samp = similarity.event_to_sample(photon_num, count, modes_dim)
         assert samp[0] == count
+
+
+class TestFeatureVectorSampling:
+    """Tests for the function ``strawberryfields.apps.graph.similarity.feature_vector_sampling``"""
+
+    def test_bad_event_photon_numbers(self):
+        """Test if function raises a ``ValueError`` when input a minimum photon number that is
+        below zero."""
+        with pytest.raises(ValueError, match="Cannot request events with photon number below zero"):
+            similarity.feature_vector_sampling([[1, 1, 0], [1, 0, 1]], [-1, 4], 1)
+
+    def test_bad_max_count(self):
+        """Test if function raises a ``ValueError`` when input a non-positive value for the
+        maximum photon count per mode."""
+        with pytest.raises(ValueError, match="Maximum number of photons per mode must be at least"):
+            similarity.feature_vector_sampling([[1, 1, 0], [1, 0, 1]], [2, 4], 0)
+
+    def test_correct_distribution(self, monkeypatch):
+        """Test if function correctly constructs the feature vector corresponding to some hard
+        coded samples. This test uses a set of samples, corresponding events, and resultant
+        feature vector to test against the output of ``feature_vector_sampling``. The
+        ``sample_to_event`` function called within ``feature_vector_sampling`` is monkeypatched
+        to return the hard coded events corresponding to the samples."""
+        samples_events_mapping = {  # max_count_per_mode = 1
+            (1, 1, 0, 0, 0): 2,
+            (1, 1, 1, 0, 0): 3,
+            (1, 1, 1, 1, 0): 4,
+            (1, 1, 1, 1, 1): 5,
+            (2, 0, 0, 0, 0): None,
+            (3, 0, 0, 0, 0): None,
+            (4, 0, 0, 0, 0): None,
+            (5, 0, 0, 0, 0): None,
+            (0, 1, 1, 0, 0): 2,
+        }
+        samples = list(samples_events_mapping.keys()) + [(1, 1, 1, 1, 1)]  # add a repetition
```

Comment:
Does this serve a specific purpose? Or just having fun with dictionaries?

Reply:
Oh, it's to have 10 samples?

Reply:
Yes, to have 10 samples overall with 1 repetition - the dictionary lets us easily get the event from the sample (used for monkeypatching in line 236), but dictionaries don't support repeated elements so I thought to make a list from the keys and add on a repetition.